site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -78,60 +98,79 @@ def __init__(self, inspire_metadata):
-                    rdm_additional_titles.append(
-                        {
-                            "title": inspire_title.get("subtitle"),
-                            "type": {
-                                "id": "subtitle",
-                            },
-                        }
-                    )
+                    subtitle = {
+                        "title": inspire_title.get("subtitle"),
+                        "type": {
+                            "id": "subtitle",
+                        },
+                    }
+                    current_app.logger.info(f"Subtitle mapped: {subtitle}.")
+                    rdm_additional_titles.append(subtitle)
             except Exception as e:
-                self.metadata_errors.append(
-                    f"Error occurred while mapping titles. Title from INSPIRE: {inspire_title}. INSPIRE record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
-                )
+                error_string = f"Error occurred while mapping titles. Title from INSPIRE: {inspire_title}. INSPIRE record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
+                current_app.logger.error(error_string)
+                self.metadata_errors.append(error_string)
                 return None, None
 
+        current_app.logger.info(
+            f"Titles transformation finished. Resulting RDM title: {rdm_title}, resulting additional titles: {rdm_additional_titles}."
+        )
         return rdm_title, rdm_additional_titles
 
     def _transform_publication_date(self):
         """Mapping of INSPIRE thesis_info.date to metadata.publication_date."""
+        current_app.logger.info("Start transforming publication date.")
         thesis_date = self.inspire_metadata.get("thesis_info", {}).get(
             "date"
         ) or self.inspire_metadata.get("thesis_info", {}).get("defense_date")
+
+        current_app.logger.debug(
+            f"Publication date from INSPIRE: {thesis_date}. Parsing to EDTF level 0 format."
+        )
         try:
             parsed_date = str(parse_edtf(thesis_date))
+            current_app.logger.info(
+                f"Publication date transformation finished. Resulting parsed value: {parsed_date}."
+            )
             return parsed_date
         except ParseException as e:
-            self.metadata_errors.append(
-                f"Error occurred while parsing imprint.date to EDTF level 0 format for publication_date. INSPIRE "
-                f"record id: {self.inspire_metadata.get('control_number')}. Date: {thesis_date}. Error: {e}."
+            error_string = (
+                f"Error occurred while parsing imprint.date to EDTF level 0 format for publication_date. "
+                f"INSPIRE record id: {self.inspire_metadata.get('control_number')}. Date: {thesis_date}. "
+                f"Error: {e}."
             )
+            current_app.logger.error(error_string)
```

Comment:
now I am thinking... if there are errors on each transformation, we should raise them to parent method and log only once there, displaying all the errors at once, since you append them to one variable anyways.
This way, the error logs reporting will be concise in one place, we won't have to scroll and search to find them one by one